santoku/sql/mysql_handler.py

```diff
@@ -0,0 +1,156 @@
+from typing import List, Tuple, Dict, Any
+
+import pandas as pd
+import mysql.connector
+
+from santoku.aws.secrets_manager_handler import SecretsManagerHandler
+
+
+class MySQLHandler:
+    """
+    Manage MySQL interactions, the simplest of which is to query a particular table.
+    This makes use of the official MySQL connector. More information at
+    https://dev.mysql.com/doc/connector-python/en/
+
+    """
+
+    def __init__(self, user: str, password: str, host: str, database: str) -> None:
+        """
+        Base constructor. Use this to connect to the MySQL server.
+        For alternative methods of authentication use the given class methods.
+
+        Parameters
+        ----------
+        user : str
+            The MySQL user name.
+        password : str
+            The password of the MySQL user.
+        host : str
+            Hostname or location of the MySQL server.
+        database : str
+            The database in the MySQL server to be connected to.
+
+        Raises
+        ------
+        mysql.connector.errors.ProgrammingError
+            Raised if credentials are not correctly specified or the database does not exsit.
+
+        mysql.connector.errors.InterfaceError
+            Raised if the connection to the MySQL server host cannot be done.
+
+        Notes
+        -----
+        More information on MySQL connection: [1].
+
+        References
+        ----------
+        [1] :
+        https://dev.mysql.com/doc/connector-python/en/connector-python-example-connecting.html
+
+        """
+        self.mysql_cnx = mysql.connector.connect(
+            user=user, password=password, host=host, database="database"
+        )
+
+    @classmethod
+    def from_aws_secrets_manager(
+        cls, secret_name: str,
+    ):
+        """
+        Retrieve the necessary information for the connection to MySQL from AWS Secrets Manager.
+        Requires that AWS credentials with the appropriate permissions are located somewhere on the
+        AWS credential chain in the local machine.
+
+        Parameters
+        ----------
+        secret_name : str
+            Name or ARN for the secret containing fields needed for MySQL authentication.
+
+        See Also
+        --------
+        __init__ : this method calls the constructor.
+
+        Notes
+        -----
+        The retrieved secret must have the following particular JSON structure:
+        ```
+        {
+            "user": "<user>",
+            "password": "<password>",
+            "host": "<host>",
+            "database": "<database>",
```

Comment:
Missing a bracket. Also, it would be appropriate to specify the database manually. We probably want the user to be able to connect to several db within the same host by reusing a secret with usr/pwd/host instead of having several almost identical secrets.